src/app-layer-ssl.c

```diff
@@ -143,11 +143,97 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            initial_input += 2;
```

Comment:
the moving forward of initial_input is opposite of what I'd expect. I'd expect input to move forward, and initial_input to be a reference to the original input, so that we can validate how far we are.